### PR TITLE
feat(sandbox): Add an example of MFA authentication in sandbox

### LINKED ISSUE #1973: Add an example of MFA authentication in sandbox
**Is your feature request related to a problem? Please describe.**
Add an example of MFA authentication in sandbox

### PR DESCRIPTION
**Pull request by arpit1503khanna**
## Description
Add an example of MFA authentication in sandbox

Fixes #1973 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by yeshamavani** [comment]
@arpit1503khanna  -- https://sonarcloud.io/project/issues?resolved=false&pullRequest=2002&id=sourcefuse_loopback4-microservice-catalog&open=AY3LsraK7m1cqhCYw4hH

File: sandbox/auth-mfa-google-example/.env.example [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+DB_PASSWORD=
+DB_DATABASE=
+DB_SCHEMA=
+REDIS_HOST=
+REDIS_PORT=
+REDIS_URL=
+REDIS_PASSWORD=
+REDIS_DATABASE=
+JWT_SECRET=
+JWT_ISSUER=
```
Comment (yeshamavani) [inline sandbox/auth-mfa-google-example/.env.example:17]: if any extra env needed please add

File: sandbox/auth-mfa-google-example/src/application.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    this.component(AuthorizationComponent);
+
+    // Set up default home page
+    this.static('/', path.join(__dirname, '../public'));
+
+    // Customize @loopback/rest-explorer configuration here
+    this.configure(RestExplorerBindings.COMPONENT).to({
+      path: '/explorer',
+    });
+    /////// My Additions
```
Comment (yeshamavani) [inline sandbox/auth-mfa-google-example/src/application.ts:115]: better comment

File: sandbox/auth-mfa-google-example/src/application.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    this.bind(AuthServiceBindings.MfaConfig).to({
+      secondFactor: STRATEGY.OTP,
+    });
+    this.bind(AuthServiceBindings.OtpConfig).to({
+      method: OtpMethodType.GOOGLE_AUTHENTICATOR,
+    });
+
+    this.bind(VerifyBindings.MFA_PROVIDER).toProvider(MfaProvider);
+
+    this.bind(Strategies.Passport.LOCAL_STRATEGY_FACTORY.key).toProvider(
```
Comment (yeshamavani) [inline sandbox/auth-mfa-google-example/src/application.ts:125]: why local strategy needed

File: sandbox/auth-mfa-google-example/src/application.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+      LocalPasswordStrategyFactoryProvider,
+    );
+    this.bind(
+      Strategies.Passport.CLIENT_PASSWORD_STRATEGY_FACTORY.key,
+    ).toProvider(ClientPasswordStrategyFactoryProvider);
+
+    this.bind(Strategies.Passport.BEARER_STRATEGY_FACTORY.key).toProvider(
+      BearerStrategyFactoryProvider,
+    );
```
Comment (yeshamavani) [inline sandbox/auth-mfa-google-example/src/application.ts:134]: with the latest version of loopback4-authentication this should not be needed please check and confirm

File: sandbox/auth-mfa-google-example/src/opentelemetry-registry.ts
Comment (yeshamavani) [inline sandbox/auth-mfa-google-example/src/opentelemetry-registry.ts:1]: ??

File: sandbox/auth-mfa-google-example/src/providers/mfa.provider.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import {Provider} from '@loopback/context';
+import {MfaCheckFn} from '@sourceloop/authentication-service';
+// import { AuthUser } from '@sourceloop/authentication-service/dist/modules/auth';
+// import { AuthUser } from '@sourceloop/authentication-service';
```
Comment (yeshamavani) [inline sandbox/auth-mfa-google-example/src/providers/mfa.provider.ts:4]: remove comments

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: sandbox/auth-mfa-google-example/src/application.ts (hunk 3 above)
Comment (arpit1503khanna) [reply sandbox/auth-mfa-google-example/src/application.ts:125]: The curent configuration requires to give a local startegy

File: sandbox/auth-mfa-google-example/.env.example (hunk 1 above)
Comment (arpit1503khanna) [reply sandbox/auth-mfa-google-example/.env.example:17]: Not in this example

File: sandbox/auth-mfa-google-example/src/application.ts (hunk 2 above)
Comment (arpit1503khanna) [reply sandbox/auth-mfa-google-example/src/application.ts]: Done

File: sandbox/auth-mfa-google-example/src/application.ts (hunk 4 above)
Comment (arpit1503khanna) [reply sandbox/auth-mfa-google-example/src/application.ts]: Yes, not needed

File: sandbox/auth-mfa-google-example/src/application.ts (hunk 3 above)
Comment (yeshamavani) [reply sandbox/auth-mfa-google-example/src/application.ts:123]: which API needed this?

File: sandbox/auth-mfa-google-example/src/application.ts [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    this.bind(Strategies.Passport.OTP_VERIFIER.key).toProvider(
+      GoogleAuthenticatorVerifyProvider,
+    );
+
+    this.bind(AuthServiceBindings.MfaConfig).to({
+      secondFactor: STRATEGY.OTP,
+    });
+    this.bind(AuthServiceBindings.OtpConfig).to({
+      method: OtpMethodType.OTP,
```
Comment (yeshamavani) [inline sandbox/auth-mfa-google-example/src/application.ts:138]: shouldnt this be google?
why binded twice

**Review by yeshamavani — CHANGES_REQUESTED** [review]

**Comment by yeshamavani** [comment]
Cant be merge it.
MFA strategy needs changes google and otp

**Pull request closed without merging**